controller/geoCodingController.js

```diff
@@ -0,0 +1,50 @@
+
+const got = require('got');
+const urlencode = require('urlencode');
+
+const geoCodingController = {
+  decode: async function (street, postal_code, city) {
+
+    if (street == undefined) {
+      throw "Street must be defined";
+    }
+
+    if (postal_code == undefined) {
+      throw "Postal code must be defined";
+    }
+
+    if (city == undefined) {
+      throw "City must be defined";
+    }
+
+    const accessToken = process.env.LOCATION_IQ_ACCESS_TOKEN;
+    const address = urlencode(street + " " + postal_code + " " + city);
+    const url = "https://us1.locationiq.com/v1/search.php?key=" + accessToken + "&q=" + address + "&format=json"
```

Comment:
This controller asks some random LocationIQ service for some geocoding. I wanted to use open source but since this is taking server power there's no free service. But this one is based on OpenStreetMaps at least.